web/src/components/wizard/config/ConfigurationStep.tsx

```diff
@@ -77,6 +83,103 @@ const ConfigurationStep: React.FC<ConfigurationStepProps> = ({ onNext }) => {
   // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  // Helper function to find the first field with field error in DOM order
+  const findFirstFieldWithError = (fieldErrors: Record<string, string>): AppConfigItem | null => {
+    if (!appConfig?.groups || Object.keys(fieldErrors).length === 0) {
+      return null;
+    }
+    
+    // Iterate through groups and items in DOM order to find first field with error
+    for (const group of appConfig.groups) {
+      for (const item of group.items) {
+        if (fieldErrors[item.name]) {
+          return item;
+        }
+      }
+    }
+    
+    return null;
+  };
+
+  // Helper function to find which group/tab contains a specific field
+  const findGroupForField = (fieldName: string): AppConfigGroup | null => {
+    if (!appConfig?.groups) return null;
+    
+    return appConfig.groups.find(group => 
+      group.items.some(item => item.name === fieldName)
+    ) || null;
+  };
+
+  // Helper function to focus on a field with tab switching support
+  const focusFieldWithTabSupport = (fieldItem: AppConfigItem): void => {
+    const targetGroup = findGroupForField(fieldItem.name);
+    
+    if (!targetGroup) {
+      console.warn(`Could not find group for field: ${fieldItem.name}`);
+      return;
+    }
+    
+    // Switch to the correct tab if field is in a different tab
+    if (targetGroup.name !== activeTab) {
+      setActiveTab(targetGroup.name);
+    }
+    
+    // Focus the field after a brief delay to ensure DOM updates complete
+    // This is especially important when switching tabs
+    const focusDelay = targetGroup.name !== activeTab ? FOCUS_DELAY_CROSS_TAB_MS : FOCUS_DELAY_SAME_TAB_MS;
+    setTimeout(() => {
+      let fieldElement: HTMLElement | null = null;
+      
+      // Special handling for file inputs - focus the upload button instead of hidden input
+      if (fieldItem.type === 'file') {
+        // For file inputs, focus the upload button using data-testid (follows pattern: file-input-{name}-button)
+        fieldElement = document.querySelector(`[data-testid="file-input-${fieldItem.name}-button"]`) as HTMLElement;
```

Comment:
we shouldn't use data test ids in the actual component. also, query by id or selector isn't how it's typically done in react, i'd suggest using refs like so:
```js
// Holds refs to each field by name
const itemRefs = useRef<Record<string, HTMLElement | null>>({});

// Assigns a ref to the element dynamically
const setRef = (name: string) => (el: HTMLElement | null) => {
  itemRefs.current[name] = el;
};

// Usage when rendering:
<input
  key={field.name}
  id={field.name}
  ref={setRef(field.name)}
  type={field.type}
/>

// To focus later:
const focusItem = (name: string) => {
  const el = itemRefs.current[name];
  if (el) {
    el.focus();
    el.scrollIntoView({ behavior: "smooth", block: "center" });
  }
};

```

Reply:
ran into some strangeness getting file items to focus correctly. let me know what you think of the implementation